packages/store/src/workspace/page.ts

```diff
@@ -471,28 +471,36 @@ export class Page extends Space<PageData> {
   insertBlock(
     blockProps: Partial<BaseBlockModel>,
     targetModel: BaseBlockModel,
-    top = true
+    top = true,
+    autofocus = false
   ) {
     const targetParentModel = this.getParent(targetModel);
     if (targetParentModel === null) {
       throw new Error('cannot find parent model');
     }
+    let id: string | null = null;
     this.transact(() => {
       const yParent = this._yBlocks.get(targetParentModel.id) as YBlock;
       const yChildren = yParent.get('sys:children') as Y.Array<string>;
       const targetIdx = yChildren
         .toArray()
         .findIndex(id => id === targetModel.id);
       assertExists(blockProps.flavour);
-      this.addBlockByFlavour(
+      id = this.addBlockByFlavour(
         blockProps.flavour,
         {
           type: blockProps.type,
         },
         targetParentModel.id,
         top ? targetIdx : targetIdx + 1
       );
-      // }
+    });
+    if (!autofocus) return;
+    // borrowed from asyncFocusRichText() in @blocksuite/blocks
+    requestAnimationFrame(() => {
```

Comment:
We shouldn't do this kind of UI mutation in the `store` package. Can you try to move the `insertBlock` method to the `blocks` package?

Reply:
Let me try to refactor here a bit.

Reply:
Thanks！！I will take care next time